src/components/Tabs/components/Tab/components/DuplicateModal/DuplicateModal.vue

```diff
@@ -0,0 +1,129 @@
+<template lang="pug">
+Modal(
+  :instant="true",
+  :open="open",
+  :primaryAction="primaryAction",
+  :secondaryActions="secondaryActions",
+  @close="emits('close')",
+)
+  template(#title) {{ i18n.translate('Polaris.Tabs.DuplicateModal.title') }}
+  ModalSection
+    Form(@submit="handlePrimaryAction")
+      FormLayout
+        div(ref="container")
+          TextField(
+            v-model="model",
+            autoComplete="off",
+            :label="i18n.translate('Polaris.Tabs.DuplicateModal.label')",
+            :helpText="helpText",
+            :maxLength="MAX_VIEW_NAME_LENGTH",
+            :showCharacterCount="true",
+            :error="errorMessage",
+            @change="handleChange",
+          )
+</template>
+
+<script setup lang="ts">
+import { ref, computed, watch } from 'vue';
+import useI18n from '@/use/useI18n';
+import { focusFirstFocusableNode } from '@/utilities/focus';
+import { Modal, Form, ModalSection, FormLayout, TextField } from '@/components';
+
+export interface DuplicateModalProps {
+  open: boolean;
+  isModalLoading?: boolean;
+  name: string;
+  helpText?: string;
+  viewNames?: string[];
+}
+
+const MAX_VIEW_NAME_LENGTH = 40;
+
+const props = defineProps<DuplicateModalProps>();
+
+const emits = defineEmits<{
+  (e: 'close'): void;
+  (e: 'click-primary-action', value: string): Promise<boolean>;
+  (e: 'click-secondary-action'): void;
+}>();
+
+const i18n = useI18n();
+const model = defineModel<string>({
+  default: '',
+});
+const container = ref<HTMLDivElement | null>(null);
+
+const hasSameNameError = computed(() => props.viewNames?.some(
+  (viewName) => viewName.trim().toLowerCase() === model.value.trim().toLowerCase(),
+));
+
+const isPrimaryActionDisabled = computed(() =>
+  props.isModalLoading ||
+  hasSameNameError.value ||
+  !model.value ||
+  model.value.length > MAX_VIEW_NAME_LENGTH,
+);
+
+const primaryAction = computed(() => ({
+  content: i18n.translate('Polaris.Tabs.DuplicateModal.create'),
+  onAction: handlePrimaryAction,
+  disabled: isPrimaryActionDisabled.value,
+}));
+
+const secondaryActions = computed(() => [
+  {
+    content: i18n.translate('Polaris.Tabs.DuplicateModal.cancel'),
+    onAction: handleSecondaryAction,
+  },
+]);
+
+const errorMessage = computed(() =>
+  hasSameNameError.value
+    ? i18n.translate(
+      'Polaris.Tabs.DuplicateModal.errors.sameName',
+      { name: model.value },
+    )
+    : undefined,
+);
+
+watch(
+  () => props.open,
+  (newVal) => {
+    if (!container.value)
+      return;
+
+    if (newVal) {
+      focusFirstFocusableNode(container.value);
+    }
+  },
+);
+
+watch(
+  () => [props.name, props.open],
+  () => {
+    if (props.open) {
+      model.value = props.name.slice(0, MAX_VIEW_NAME_LENGTH);
```

Comment:
Is this for truncated? 
So the model value should be get/set computed or it will be conflicted with the logic below. (comment underneath)